src/components/CommitBox.tsx

```diff
@@ -87,7 +98,11 @@ export class CommitBox extends React.Component<
       ? 'Disabled: No files are staged for commit'
       : !this.state.summary
       ? 'Disabled: No commit message summary'
-      : 'Commit';
+      : !this.props.useCommitAndPush
+      ? 'Commit'
+      : !this.props.branches.some(branch => branch.is_remote_branch) 
+      ? 'Disabled: No remote repository defined'
+      : 'Commit and push';
```

Comment:
As the `onCommit` action comes from a property, this component is blind to the actions taking place. So a better design is to add the button label as property of the `CommitBox` and handle the logic in `GitPanel`.